src/main/java/TasksManager.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+
+public class TasksManager {
+    private static final int EVENT_DEADLINE_TASK_NUMBER_OF_PARAMETERS = 2;
+
+    private ArrayList<Task> list;
+    private int numberOfTasks;
+
+    public TasksManager() {
+        this.list = new ArrayList<Task>();
+        this.numberOfTasks = 0;
+    }
+
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    public boolean isEmpty() {
+        return list.isEmpty();
+    }
+
+    public int getNumberOfTasks() {
+        return this.numberOfTasks;
+    }
+
+    public void setNumberOfTasks(int numberOfTasks) {
+        this.numberOfTasks = numberOfTasks;
+    }
+
+    public void incrementNumberOfTasks() {
+        this.numberOfTasks += 1;
+    }
+
+    public boolean addTask(String text) {
+        // Create new Task object with text
+        boolean taskAddedSuccessfully;
+        String taskType = extractTaskType(text);
+
+        switch (taskType) {
+        case SoraUI.ADD_TODO_COMMAND_KEYWORD:
+            String todoDescription = removeCommandKeyword(text);
+            taskAddedSuccessfully = list.add(new Todo(todoDescription));
+            break;
+        case SoraUI.ADD_EVENT_COMMAND_KEYWORD:
+            String[] eventDescriptionAndDate = extractDescriptionAndDate(text, SoraUI.ADD_EVENT_OPTION_KEYWORD);
+            taskAddedSuccessfully = list.add(new Event(eventDescriptionAndDate));
+            break;
+        case SoraUI.ADD_DEADLINE_COMMAND_KEYWORD:
+            String[] deadlineDescriptionAndDate = extractDescriptionAndDate(text, SoraUI.ADD_DEADLINE_OPTION_KEYWORD);
+            taskAddedSuccessfully = list.add(new Deadline(deadlineDescriptionAndDate));
+            break;
+        default:
+            taskAddedSuccessfully = false;
+        }
+
+        if (taskAddedSuccessfully) {
+            incrementNumberOfTasks();
+            return true;
+        }
+
+        return false;
+    }
+
+    private String extractTaskType(String text) {
+        String taskType = text.toLowerCase().split(" ", 2)[0];
+        return taskType;
+    }
+
+    private String removeCommandKeyword(String text) {
+        String description = text.split(" ", 2)[1];
+        return description;
+    }
+
+    private String[] extractDescriptionAndDate(String text, String optionKeyword) {
+        String taskDescriptionAndDateString = removeCommandKeyword(text);
+        String[] taskDescriptionAndDate = new String[EVENT_DEADLINE_TASK_NUMBER_OF_PARAMETERS];
+
+        for (int i = 0; i < taskDescriptionAndDate.length; i += 1) {
+            taskDescriptionAndDate[i] = taskDescriptionAndDateString.split(optionKeyword)[i].trim();
```

Comment:
perhaps can split this into 2 lines, 1 for split 1 for trim to boost readability